src/auth/awsclient.ts

```diff
@@ -335,4 +337,21 @@ export class AwsClient extends BaseExternalAccountClient {
     });
     return response.data;
   }
+
+  private shouldUseMetadataServer(): boolean {
+    return (
+      !this.canRetrieveRegionFromEnvironment() ||
+      !this.canRetrieveSecurityCredentialsFromEnvironment()
+    );
+  }
+
+  private canRetrieveRegionFromEnvironment(): boolean {
+    return !!(process.env['AWS_REGION'] || process.env['AWS_DEFAULT_REGION']);
+  }
```

Comment:
This method would be more useful as a getter than a boolean, and we should use it wherever we currently use this check.

```suggestion
  private get regionFromEnvironment(): string | undefined {
    return process.env['AWS_REGION'] || process.env['AWS_DEFAULT_REGION'];
  }
```